### PR TITLE
fix(debug): allow debug after m365 account swtiched

### PR DESCRIPTION
**Pull request by kuojianlu**
@microsoft/teamsfx-cli	v1.1.3-rc.2
@microsoft/teamsfx-core	v1.17.0-rc.2
ms-teams-vscode-extension	v4.1.0-rc.2

CLI BREAKING CHANGE:
  

Extension-toolkit BREAKING CHANGE:
 

CLI feat commits:
  

Extension-toolkit feat commits:

### CODE REVIEW

**Comment by kuojianlu** [comment]
Fix https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/15872897.

E2E TEST: N/A

**Review by XiaofuHuang — APPROVED** [review]

File: packages/fx-core/src/component/debugHandler/utils.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    logger,
+    telemetry,
+    ui,
+    cryptoProvider
+  );
+  const inputs = constructInputsWithProjectPath(projectPath);
+  const tenantId =
+    envInfoV3.state[ComponentNames.AadApp]?.tenantId ||
+    envInfoV3.state[ComponentNames.AppManifest]?.tenantId;
+  const checkResult = await checkWhetherLocalDebugM365TenantMatches(
```
Comment (swatDong) [inline packages/fx-core/src/component/debugHandler/utils.ts:52]: Is there the case - user does not login, and run the task(s) directly. Will this ask user to login again and again?

Comment (kuojianlu) [reply packages/fx-core/src/component/debugHandler/utils.ts:52]: `checkWhetherLocalDebugM365TenantMatches` does ask user to login if user has not loged in.  The tasks are running in sequence, so only the first time login is required, and the following steps will run as normal.

Comment (swatDong) [reply packages/fx-core/src/component/debugHandler/utils.ts:52]: I mean, what if use cancels login? Or that's out of scope?

Comment (kuojianlu) [reply packages/fx-core/src/component/debugHandler/utils.ts:52]: If user cancels login, there will be an error to terminate the running task.
image

**Review by swatDong — APPROVED** [review]

**Pull request merged**